runtime/eventlog.c

```diff
@@ -41,32 +41,80 @@
 
 #define CTF_MAGIC 0xc1fc1fc1
 #define CAML_TRACE_VERSION 0x1
+#define CTF_METADATA_MAGIC 0x75D11D57
+#define CTF_MAJOR_VERSION 1
+#define CTF_MINOR_VERSION 8
+#define MIN_AVAILABLE_EVENT_ID 5
+#define CTF_TRACE_UUID {0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa, 0xaa}
 
+/* The structs in this section are emitted structs, i.e. they
+are emitted and writen to the files during the trace */
+
+/* Header at the beginning of every packet in a stream */
 struct ctf_stream_header {
   uint32_t magic;
   uint16_t caml_trace_version;
+  uint8_t uuid[16];
   uint16_t stream_id;
 };
 
+/* Header for the eventlog 'data' stream */
 static struct ctf_stream_header header = {
   CTF_MAGIC,
   CAML_TRACE_VERSION,
+  CTF_TRACE_UUID,
   0
 };
 
+/* Special header for the 'metadata' stream */
+#pragma pack(1)
```

Comment:
This is the MSVC syntax for packing, and maybe surprisingly it's global - this affects all structs in the file.

If you just want to `pack` a single structure, you can use `pragma pack(push)` and `pragma pack(pop)` around the structure (see https://docs.microsoft.com/en-us/cpp/preprocessor/pack?view=vs-2019).

But it's probably simpler to use the GCC syntax instead (we can worry about porting to Windows later), and write
```
struct __attribute__((packed)) ctf_metadata_header {
  ...
};
```

Reply:
Oh hm, that's interesting. There is already a `#pragma pack(1)` that occurs on line 328 so should I get rid of both of them or should I just keep (one of) them since it was already present?